src/include/cell_inline.h

```diff
@@ -53,24 +53,42 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
     ++*pp;
 
     flags = 0;
-    if (tw->start_ts != WT_TS_NONE) {
+    const bool pack_prepare_info_to_start = tw->prepare && tw->prepare_ts != WT_TS_NONE && (tw->stop_txn == WT_TXN_MAX || tw->stop_txn == tw->start_txn);
+    const bool pack_prepare_info_to_stop = tw->prepare && tw->prepare_ts != WT_TS_NONE && tw->stop_txn != WT_TXN_MAX;
```

Comment:
This can be

```
const bool pack_prepare_info_to_start = F_ISSET(btree, WT_BTREE_DISAGGREGATED) && tw->prepare && tw->prepare_ts != WT_TS_NONE && (!WT_TIME_WINDOW_HAS_STOP(tw) || tw->stop_txn == tw->start_txn);
const bool pack_prepare_info_to_stop = F_ISSET(btree, WT_BTREE_DISAGGREGATED) && tw->prepare && tw->prepare_ts != WT_TS_NONE && WT_TIME_WINDOW_HAS_STOP(tw)

Reply:
fixed